scipy/stats/tests/test_correlation.py

```diff
@@ -23,16 +27,17 @@ def test_against_R_XICOR(self, case):
         #       0.7620801065573549, 0.31410063302647495, 0.7935620302236199,
         #       0.5423085761365468)
         # xicor(x, y, ties=FALSE, pvalue=TRUE)
-
+        dtype = xp_default_dtype(xp) if dtype is None else getattr(xp, dtype)
         rng = np.random.default_rng(25982435982346983)
         x = rng.random(size=10)
-
         y = (rng.random(size=10) if case['y_cont']
              else rng.integers(0, 5, size=10))
+
+        x, y = xp.asarray(x, dtype=dtype), xp.asarray(y, dtype=dtype)
         res = stats.chatterjeexi(x, y, y_continuous=case['y_cont'])
 
-        assert_allclose(res.statistic, case['statistic'])
-        assert_allclose(res.pvalue, case['pvalue'])
+        xp_assert_close(res.statistic, xp.asarray(case['statistic'], dtype=dtype))
+        xp_assert_close(res.pvalue, xp.asarray(case['pvalue'], dtype=dtype))
 
     @pytest.mark.parametrize('y_continuous', (False, True))
     def test_permutation_asymptotic(self, y_continuous):
```

Comment:
This can get converted when gh-23772 is merged, but I don't think that needs to hold up this PR.